packages/core/src/external/sftp/surescripts/schema/verification.ts

```diff
@@ -0,0 +1,236 @@
+    fromSurescripts: fromSurescriptsDate(),
+  },
+  {
+    field: 6,
+    fromSurescripts: fromSurescriptsTime(),
+  },
+  {
+    field: 7,
+    key: "transactionFileType",
+    fromSurescripts: fromSurescriptsEnum(["PMA"]),
+  },
+  {
+    field: 8,
+    key: "transmissionControlNumber",
+    fromSurescripts: fromSurescriptsString(),
+  },
+  {
+    field: 9,
+    key: "transmissionDate",
+    fromSurescripts: fromSurescriptsDate(),
+  },
+  {
+    field: 10,
+    fromSurescripts: fromSurescriptsTime(),
+  },
+  {
+    field: 11,
+    key: "usage",
+    fromSurescripts: (value: string) => {
+      switch (value) {
+        case "T":
+          return "test";
+        case "P":
+          return "production";
+        default:
+          throw new Error(`Invalid file type: ${value}`);
+      }
+    },
+  },
+  {
+    field: 12,
+    key: "loadStatus",
+    fromSurescripts: (value: string) => {
+      switch (value) {
+        case "01":
+          return "success";
+        case "02":
+          return "partial_success";
+        case "03":
+          return "failure";
+        default:
+          throw new Error(`Invalid load status: ${value}`);
+      }
+    },
+  },
+  {
+    field: 13,
+    key: "loadStatusDescription",
+    fromSurescripts: fromSurescriptsString(),
+  },
+];
+
+export const patientVerificationDetailSchema = z.object({
+  recordType: z.enum(["SDT"]),
+  recordSequenceNumber: z.number(),
+  sourceRecordSequenceNumber: z.number().optional(),
+  assigningAuthority: z.string().min(1).max(64).optional(),
+  patientId: z.string().min(1).max(35),
+  errorType: z.enum(["warning", "error", "fatal"]),
+  errorCode: z.string().min(1).max(10),
+  errorDescription: z.string().min(1).max(250),
+});
+
+export type PatientVerificationDetail = z.infer<typeof patientVerificationDetailSchema>;
+
+export function isPatientVerificationDetail(data: object): data is PatientVerificationDetail {
+  return patientVerificationDetailSchema.safeParse(data).success;
+}
+
+export const patientVerificationDetailOrder: IncomingFileRowSchema<PatientVerificationDetail> = [
+  {
+    field: 0,
+    key: "recordType",
+    fromSurescripts: fromSurescriptsEnum(["SDT"]),
+  },
+  {
+    field: 1,
+    key: "recordSequenceNumber",
+    fromSurescripts: fromSurescriptsInteger(),
+  },
+  {
+    field: 2,
+    key: "sourceRecordSequenceNumber",
+    fromSurescripts: fromSurescriptsIntegerOrUndefined(),
+  },
+  {
+    field: 3,
+    key: "assigningAuthority",
+    fromSurescripts: fromSurescriptsString(),
+  },
+  {
+    field: 4,
+    key: "patientId",
+    fromSurescripts: fromSurescriptsString(),
+  },
+  {
+    field: 5,
+    key: "errorType",
+    fromSurescripts: (value: string) => {
+      switch (value) {
+        case "W":
+          return "warning";
+        case "E":
+          return "error";
+        case "F":
+          return "fatal";
+        default:
+          throw new Error(`Invalid error type: ${value}`);
+      }
+    },
```

Comment:
I suggest we always pass a reference to a function, never implement one inline, otherwise it opens a precedence for people to start writing logic in the setup of the thing.

Reply:
Totally agree! Also makes it a bit harder to reason about the spec, since some strings are being converted during the reading process - I'm making the initial TS interface align more closely with the spec, and then future transformations (e.g. mapping strings to other strings) would be in a subsequent decomposition step.